### ISSUE TITLE
react-rules-of-hooks conditional call detection is broken

### ISSUE THREAD

**Issue by andrejcremoznik** [denoland/deno_lint#1421]
### Lint Name

react-rules-of-hooks

### Code Snippet

```js
const useA = (config) => console.log(config)

const useB = ({ overrides } = {}) => {
  const config = {}
  if (overrides) Object.assign(config, overrides)
  return useA(config)
}
```

### Expected Result

No linter errors.

### Actual Result

"Hook called conditionally" error.

### Additional Info

The conditional call check is only relevant if the condition prevents the hook from executing e.g. there's a `return` before the hook call.

https://react.dev/reference/rules/rules-of-hooks

### Version

deno 2.1.10 (stable, release, x86_64-unknown-linux-gnu)
v8 13.0.245.12-rusty
typescript 5.6.2

**Comment by marvinhagemeister** [comment]
Fixed by https://github.com/denoland/deno_lint/pull/1418
